src/result.js

```diff
@@ -112,22 +112,34 @@ class Result {
   }
 }
 
-/**
- * `String.replace()` converts '$$' to '$', so we must escape each '$' as '$$';
- * but because we’re using `String.replace()` to do it, we must use '$$$'!
- */
-function escapeForStringReplace(string) {
-  return string.replace(/\$/g, '$$$');
-}
-
 function insertIntoIndexHTML(html, head, body) {
-  html = html.replace("<!-- EMBER_CLI_FASTBOOT_BODY -->", escapeForStringReplace(body));
+  if (!html) { return Promise.resolve(html); }
+
+  if (body) {
+    let isBodyReplaced = false;
+    html = html.replace("<!-- EMBER_CLI_FASTBOOT_BODY -->", function() {
+      isBodyReplaced = true;
+      return body;
+    });
+
+    if (!isBodyReplaced) {
+      return Promise.reject(new Error("`<!-- EMBER_CLI_FASTBOOT_BODY -->` `insert comment` is missing in your index.html"));
```

Comment:
You can wrap the error message in its function to avoid duplicating it in two places.